xbmc/video/windows/GUIWindowFullScreen.cpp

```diff
@@ -473,3 +474,11 @@ bool CGUIWindowFullScreen::HasVisibleControls()
 {
   return m_controlStats->nCountVisible > 0;
 }
+
+int CGUIWindowFullScreen::GetOSDWindowID()
```

Comment:
at least keep the state within the object. having the object ask: "hey application, do you know what type of a window I am?" does not look right.